src/libraries/icubmod/embObjMotionControl/eomcParser.h

```diff
@@ -56,6 +57,181 @@ namespace yarp {
 //    PidAlgo_currentInnerLoop =2
 //} PidAlgorithmType_t;
 
+class ITemperatureSensor
```

Comment:
It is a good idea to implement the "generic interface" `ITemperatureSensor`, but I think you can use the constructor instead of `setTemperatureSensor` function.
In the constructor, you can initialize directly all the specific values of gradient, offset, etc.
In this way, you don't need the parameter `isConfigured`. If the object exists it is also configured.

See the line 1419 of embObjParser.cpp

In addition all the private field could be a const and initialized in the creation pahse.